src/Workspaces/VisualBasic/Portable/Extensions/SyntaxTokenExtensions.vb

```diff
@@ -57,6 +57,33 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
         End Function
 
         <Extension()>
+        Public Function IsGlobalStatementContext(token As SyntaxToken, position As Integer) As Boolean
+            If Not token.IsLastTokenOfStatement() Then
+                Return False
+            End If
+
+            Dim isIncomplete = token.GetAncestor(Of IncompleteMemberSyntax)() IsNot Nothing
```

Comment:
While it is generally preferable to defer computation until is is required (in this case, until we know that `token.HasTrailingTrivia` is true), this code is less performance critical than compiler code.